plugins/zynaddsubfx/zynaddsubfx/src/Misc/QtXmlWrapper.cpp

```diff
@@ -193,9 +202,18 @@ int QtXmlWrapper::dosavefile(const char *filename,
                            int compression,
                            const char *xmldata) const
 {
+    QFile f(filename);
+    f.open(QIODevice::WriteOnly);
+#ifdef LMMS_BUILD_WIN32
+    int fd = _dup(f.handle());
+#else
+    int fd = dup(f.handle());
+#endif
+    f.close();
```

Comment:
I see you're using this to open files by their descriptor instead of the filename. Is it `gzopen` that's giving you problems, `fopen`, or both? Because if it's only `gzopen` that's broken, it might be possible to just always `fopen`, and then call `gzdopen(fileno(file), options)` if compression is enabled, rather than doing this trickery with duplicating the file handles and such.

Reply:
Both are broken, because C library only supports characters in local code page.

Reply:
Oh, oops. I had typed out that comment and after digging deeper to better understand the problem I meant to remove it before posting the review >.<

Yeah, I agree with your conclusion here. It's not pretty, but I don't see any better solution.